train/trainer.py

```diff
@@ -254,27 +260,58 @@ def reconfigure_optimizers(self):
         optimizers, _ = self.configure_optimizers(
             existing_scheduler=self.lr_schedulers())
         self.optimizers()._optimizer = optimizers[0]
-
+    
+    def get_optimizer_params(self, weight_decay):
+        decay_params = []
+        no_decay_params = []
+        for name, param in self.model.named_parameters():
+            if not param.requires_grad:
+                continue  # Exclude frozen parameters
+            if "bias" in name or "LayerNorm.weight" in name or "LayerNorm.bias" in name:
+                no_decay_params.append(param)
+            else:
+                decay_params.append(param)
+        return [
+            {'params': decay_params, 'weight_decay': weight_decay},
+            {'params': no_decay_params, 'weight_decay': 0.0}
+        ]
+    
     def configure_optimizers(self, existing_scheduler=None):
-        optimizer = torch.optim.Adam(self.parameters(),
-                                     lr=self.train_params.lr)
+        weight_decay = self.train_params.weight_decay
 
-        def f_warmup(n):
-            assert n <= self.train_params.lr_warm_steps
-            return n / self.train_params.lr_warm_steps
+        if weight_decay > 0:
```

Comment:
once get_optimizer_params is updated to work even for weight decay zero, can make this code neater:
```
optimizer_params = self.get_optimizer_params(weight_decay)
optimizerClass = AdamW if weight_decay > 0 else Adam
optimizer = optimizerClass(optimizer_params, lr=self.train_params.lr)
```